metadata.json

```diff
@@ -7,8 +7,8 @@
   "source": "https://github.com/voxpupuli/puppet-hosts",
   "dependencies": [
     {
-      "name": "puppetlabs/stdlib",
-      "version_requirement": ">= 8.0.0 < 10.0.0"
+      "name": "puppetlabs/host_core",
```

Comment:
please don't add the core module. They are supposed to be always available so we never add them to metadata.json.

Reply:
I deeply disagree with this. For starters, it isn't always available. It is only available when using the AIO packaging and AFAIK, there is no official vox policy of only support AIO.  `puppetlabs/host_core` is a dependency and it has a  version string.  If we don't put it in the metadata.json then we have to do extra work to setup the acceptance tests, which is just shooting ourselves by causing extra work for no benefit.

Reply:
Even non AIO packages should include those core modules or pull them in as a dependency. Debian fox examples does that: https://packages.debian.org/bookworm/puppet-agent. Same on ubuntu: https://packages.ubuntu.com/lunar/puppet-agent. Arch vendors them into the puppet package: https://gitlab.archlinux.org/archlinux/packaging/packages/puppet/-/blob/main/PKGBUILD?ref_type=heads#L38-46

because of that, we don't need to explicitly install them in our acceptance tests. If a distro builds their own package but doens't pull in the core modules, the package is broken (that's a statement from Puppet, but I agree with this).

Reply:
These are modules that have versions. They are released on the forge so they can be updated *independently* from the core.

Reply:
Let me turn this around, what is broken by this module's metadata declaring its full dependency chain?

Reply:
Also, it needs to be documented that this module has a dep on host_core as when someone copies into into the `Puppetfile` for the control repo, `host_core` also needs to be declared in the `Puppetfile` for rspec-puppet to work.

Reply:
I'm going to ship this as-is as I'm in a rush. We can continue this discussion if needed.